Add BattleStations addon

Sjshovan opened the pull request.

Sjshovan committed:
Add BattleStations addon

Sjshovan committed:
update installation instructions

z16 commented on addons/BattleStations/helpers.lua:
```diff
@@ -0,0 +1,121 @@
+end
+
+function buildHelpTypeEntry(name, description)
+	local entry_template = "%s %s %s"
```
A single tab snuck in somehow :)

Sjshovan replied:
done

z16 commented on addons/addons.xml:
```diff
@@ -86,6 +86,13 @@
     <bugtracker>https://github.com/Byrth/Lua-Byrth/issues</bugtracker>
     <support>https://discord.gg/b275nMv</support>
+  </addon>
+   <addon>
```
A single extra space snuck in, somehow :)

Sjshovan replied:
done

z16 commented on addons/BattleStations/battlestations.lua:
```diff
@@ -0,0 +1,598 @@
+        buildHelpCommandEntry('default [radio]', 'Set radio(s) to the default station (Current Zone Music).'),
+        buildHelpCommandEntry('normal [radio]', 'Set radio(s) to the original game music.'),
+        buildHelpCommandEntry('reload', 'Reload Battle Stations.'),
+		buildHelpCommandEntry('about', 'Display information about Battle Stations.'),
```
You used tabs here instead of spaces.

Sjshovan replied:
done

z16 commented on addons/BattleStations/battlestations.lua:
```diff
@@ -0,0 +1,598 @@
+        buildHelpSeperator('=', 25),
+    },
+	
+	 about = {
```
Another tab.

Sjshovan replied:
done

z16 commented on addons/BattleStations/battlestations.lua:
```diff
@@ -0,0 +1,598 @@
+function handleInjectionNeeds() 
+	if needs_inject then
+		injectBattleMusic()
+		needs_inject = false;
```
More tabs.

Sjshovan replied:
done

z16 commented on addons/BattleStations/battlestations.lua:
```diff
@@ -0,0 +1,598 @@
+end)
+
+windower.register_event('outgoing chunk', function(id, data)
+	if id == packets.outbound.action.id then
```
Another tab... it may make sense to just do a search and replace, looks like it happens occasionally.

Sjshovan replied:
done

z16 commented on addons/BattleStations/battlestations.lua:
```diff
@@ -0,0 +1,598 @@
+    local success = true
+
+    if command == 'list' or command == 'l' then
+        local list_type = tostring(command_args[1] or '*'):lower()
```
You use `tostring` around command parameters a lot, you never need those. Command parameters are always already strings.

Sjshovan replied:
done

z16 commented on addons/BattleStations/battlestations.lua:
```diff
@@ -0,0 +1,598 @@
+    if act.actor_id == windower.ffxi.get_player().id then
+        if act.category == 4 and act.recast == 225 and act.targets[1].actions[1].animation == 939 then
+            if not playerInParty() then
+                functions.loop(injectBattleMusic, 1, 5)          
```
You can write this as `injectBattleMusic:loop(1, 5)`, just fyi.

Sjshovan replied:
Thanks for the info! I will leave it as is since it is explicit this way.

z16 reviewed (approved):
Just noted down a few minor comments, most of it looks pretty good. Let me know if you want me to merge it as is or if you want to adjust some of the things I mentioned.

Sjshovan committed:
replace tabs with spaces
remove unnecessary string conversions
remove accidental space character in addons.xml

Sjshovan commented:
Thank you for the review. I believe we are ready to merge.
